coderd/apikey.go

```diff
@@ -77,8 +84,19 @@ func (api *API) postToken(rw http.ResponseWriter, r *http.Request) {
 		ExpiresAt:       database.Now().Add(lifeTime),
 		Scope:           scope,
 		LifetimeSeconds: int64(lifeTime.Seconds()),
+		TokenName:       tokenName,
 	})
 	if err != nil {
+		if database.IsUniqueViolation(err) {
```

Comment:
I think you can be even more exact here.
```suggestion
		if database.IsUniqueViolation(err, , database.UniqueIndexApiKeyName) {
```